### PR TITLE
Run tests with require:true (relative require).

### PR DESCRIPTION
**Pull request by johnjbarton**
Interpret with require:true.
Fix tests that had off-by-one directory errors.
Add test for require directory.

### CODE REVIEW

File: test/modular/TraceurTestRunner.js [hunk 1]
```diff
@@ -25,7 +25,8 @@ export class TraceurTestRunner extends Mocha6 {
       ignoreLeaks: true,
       importMetadata: {
         traceurOptions: {
-          sourceMaps: 'memory'
+          sourceMaps: 'memory',
+          require: true
```
Comment (arv) [inline test/modular/TraceurTestRunner.js]: is test/modular nodejs only?

File: test/unit/node/api.js [hunk 2]
```diff
@@ -16,7 +16,7 @@ import {suite, test, assert} from '../../unit/unitTestRunner.js';
 
 suite('api.js', function() {
 
-  var api = require('../../src/node/api');
+  var api = require('../../../src/node/api');
```
Comment (arv) [inline test/unit/node/api.js]: how did this work before?

File: src/node/require.js [hunk 3]
```diff
@@ -53,6 +54,8 @@ function shouldCompile(filename) {
   return false;
 }
 
+traceurRequire.nodeRequireJs = require;
```
Comment (arv) [inline src/node/require.js]: maybe nodeRequire or nodeJsRequire?

**Comment by arv** [comment]
How did you run into this?

File: test/unit/node/api.js (hunk 2 above)
Comment (johnjbarton) [inline test/unit/node/api.js]: Before we executed the `require()` call from within a nodejs module `/work/traceur-compiler/src/node/System.js`.  Thus the filename `.../../src/node/api` resolves correctly.

To say this another way, our test files were not node modules, just files read by our test app and executed.  Their `require()` calls thus run in the context of the test driver.

**Comment by johnjbarton** [comment]
While converting `TraceurBrowser` to `NodeTraceurBrowser` I realized our execution environment for tests was not quite what I thought it was.  In trying to clean that up I started to enforce correct module loading and then re-discovered  that we load our tests as if we where in a file in `src/node`.

File: test/modular/TraceurTestRunner.js (hunk 1 above)
Comment (johnjbarton) [inline test/modular/TraceurTestRunner.js]: Sadly it is currently, I'll fix it, #1818

**Comment by johnjbarton** [comment]
PTAL

**Comment by arv** [comment]
LGTM

**Pull request closed without merging**